docs/topics/migrations.txt

```diff
@@ -629,27 +632,67 @@ to be able to write your own, see the :doc:`migration operations reference
 </ref/migration-operations>` and the "how-to" on :doc:`writing migrations
 </howto/writing-migrations>`.
 
-.. _migration-squashing:
 
-Squashing migrations
-====================
+Optimizing migrations
+=====================
 
-You are encouraged to make migrations freely and not worry about how many you
-have; the migration code is optimized to deal with hundreds at a time without
-much slowdown. However, eventually you will want to move back from having
-several hundred migrations to just a few, and that's where squashing comes in.
+Django developers are free to create as migrations as required and do not have
+to worry about the number of migrations created as Django is optimized to deal
+with hundreds at a time without much slowdown.  However, there may come a time when migrations will become so numerous that it
+will start to have a noticeable impact on various tasks like running tests on
+CI.
 
-Squashing is the act of reducing an existing set of many migrations down to
-one (or sometimes a few) migrations which still represent the same changes.
+There are a few steps that developers can take to mitigate or remediate this by
+reducing the amount of work that migrations has to do:
+
+
+Optimizing migrations
+---------------------
 
-Django does this by taking all of your existing migrations, extracting their
-``Operation``\s and putting them all in sequence, and then running an optimizer
-over them to try and reduce the length of the list - for example, it knows
+Django will automatically optimize migrations created via the
+:djadmin:`makemigrations` command however you may also choose to optimize manually
+created migrations with the :djadmin:`optimizemigration` command:
+
+.. code-block:: shell
+
+    $ python manage.py optimizemigration <app_label> <migration_name>
+
+Migration optimization attempts to reduce the list of migration operations by
+merging known pairs of possible reductions together.  For example, it knows
 that :class:`~django.db.migrations.operations.CreateModel` and
 :class:`~django.db.migrations.operations.DeleteModel` cancel each other out,
 and it knows that :class:`~django.db.migrations.operations.AddField` can be
 rolled into :class:`~django.db.migrations.operations.CreateModel`.
 
+
+Updating migrations
+-------------------
+
+When creating new migrations you have the option of updating the previously
+created migration if one exists.  Running ``makemigrations --update`` will add any new operations to the last
+migration created, then optimize it:
+
+.. code-block:: shell
+
+    $ python manage.py makemigrations --update
```

Comment:
TIL :scream: 

Could we perhaps provide a small paragraph with an example? I can imagine what this does, but since I never tried it, I'm not completely sure what to expect as a result.